api4/status.go

```diff
@@ -15,7 +15,7 @@ func (api *API) InitStatus() {
 	api.BaseRoutes.User.Handle("/status", api.ApiSessionRequired(updateUserStatus)).Methods("PUT")
 	api.BaseRoutes.User.Handle("/status/custom", api.ApiSessionRequired(updateUserCustomStatus)).Methods("PUT")
 	api.BaseRoutes.User.Handle("/status/custom", api.ApiSessionRequired(removeUserCustomStatus)).Methods("DELETE")
-	api.BaseRoutes.User.Handle("/status/custom/recent", api.ApiSessionRequired(removeUserRecentCustomStatus)).Methods("DELETE")
+	api.BaseRoutes.User.Handle("/status/custom/recent", api.ApiSessionRequired(removeUserRecentCustomStatus)).Methods("POST")
```

Comment:
@isacikgoz Refer to the conversation here: https://community.mattermost.com/core/pl/943phgpfs7869repfjdno5ipyw

Basically, the mobile app does not allow making a `DELETE` request with a request body.

This is changed to a `POST` request similar to `deletePreferences`.
https://github.com/mattermost/mattermost-server/blob/e974b7b9be5ec499b2e48a86038b6edbb1fa8bbd/api4/preference.go#L16

Reply:
Thanks for the explanation @chetanyakan I see the rationale now. We need to be sure if this change won't break the API.

Reply:
If the feature is considered experimental, I could see breaking that API being allowable. Alternatively, we could consider supporting both HTTP methods even if that's a bit clunky.

Reply:
I see, as this rolled out with `v5.33` I'd prefer adding new endpoint for this.

Reply:
I'm 0/5 on maintaining compatibility just because it did ship in a previous release since it's experimental, but looking at this again, I think a new endpoint makes sense because having `POST /status/custom/recent` delete an entity is really counterintuitive.

Reply:
@hmhealey How about `POST /status/custom/recent/delete`?

Reply:
Yep, that should work. Maybe double check to make sure we don't have some other naming convention that we're following for non-DELETE delete APIs, but that sounds good to me otherwise.

Reply:
The only other endpoint I can find that works like that is `POST /preferences/delete` which also takes a body containing what to delete, so it matches this perfectly :+1:

Reply:
Done.